--- a/clang/lib/AST/ByteCode/Interp.cpp
+++ b/clang/lib/AST/ByteCode/Interp.cpp
@@ -41,8 +41,10 @@
 // We disable it on MSVC generally since it doesn't seem to be able
 // to handle the way we use tailcalls.
 // PPC can't tail-call external calls, which is a problem for InterpNext.
+// WebAssembly can only tail-call with the tail-call feature.
 #if defined(_MSC_VER) || defined(__powerpc__) || !defined(MUSTTAIL) ||         \
-    defined(__i386__) || defined(__sparc__)
+    defined(__i386__) || defined(__sparc__) ||                                 \
+    (defined(__wasm__) && !defined(__wasm_tail_call__))
 #undef MUSTTAIL
 #define MUSTTAIL
 #define USE_TAILCALLS 0
